packages/docusaurus-utils/src/index.ts

```diff
@@ -463,3 +463,17 @@ export function reportMessage(
       );
   }
 }
+
+export function getSwizzledComponent(
+  componentPath: string,
+): string | undefined {
+  const swizzledComponentPath = path.resolve(
+    process.cwd(),
+    'src',
```

Comment:
Due to the circular dependency issue, I can't use a constant from core (`import {SRC_DIR_NAME} from '@docusaurus/core/lib/constants';`) here, so I had to use a hard-coded string `src`.

```
lerna WARN ECYCLE Dependency cycles detected, you should fix these!
lerna WARN ECYCLE @docusaurus/utils-validation -> @docusaurus/utils -> @docusaurus/core -> @docusaurus/utils-validation
```

Reply:
yes already had this issue.

I think we could have a `@docusaurus/constants` project to solve this once for all, what do you think?

Reply:
I don't want to create new package that will include only constants, maybe we can add something else to it? But if there are no other options, then I think it's worth moving the constants into separate package.

Reply:
That's a bit annoying to have too many packages that do very little, but afaik it's the only clean solution to handle this properly.

Even in core, I had to split in server-side constants / client-side constants, and needed some duplication, see also packages/docusaurus/src/client/exports/constants.ts